data/calculated_parser/functions.py

```diff
@@ -385,20 +386,29 @@ def soundchannelaxis(depth, lat, temperature, salinity):
     """
     
     speed = sspeed(depth, lat, temperature, salinity)
-    result = np.empty((speed.shape[-2], speed.shape[-1]))
-    for x in range(speed.shape[-1]):
-        for y in range(speed.shape[-2]):
-            speed_point = speed[:,y,x]
-            if count_numerical_vals(speed_point) != 0:
-                idx = find_sca_idx(speed_point)
-                if np.isnan(idx):
-                    result[y,x] = idx
-                else:
-                    result[y,x] = depth.values[idx]
-            else:
-                result[y,x] = np.nan
-                
-    return result
+    
+    # Mask Array to allow for min index search
+    array = np.ma.masked_array(speed, np.isnan(speed))
+
+    # Find index's of minimum values
+    min_idx = array.argmin(axis = 0)
```

Comment:
Could `np.nanargmin` be used here?

https://docs.scipy.org/doc/numpy/reference/generated/numpy.nanargmin.html

Reply:
No, this returns a value error when it hits an all NaN slice